### PR TITLE
feat: added API testing support

### PR DESCRIPTION
**Pull request by hmanchev**

### CODE REVIEW

File: src/uiveri5.js [hunk 1]
```diff
@@ -336,6 +345,8 @@ function run(config) {
               return;
             }
 
+            console.debug("Spec object: " + JSON.stringify(spec));
```
Comment (hmanchev) [inline src/uiveri5.js]: To be deleted.

File: src/uiveri5.js [hunk 2]
```diff
@@ -217,6 +217,13 @@ function run(config) {
           comparisonProvider.register(matchers);
         }
 
+        var apiTestingMatchers = moduleLoader.loadModuleIfAvailable('matchers', []);
```
Comment (maximnaidenov) [inline src/uiveri5.js]: lets use a more generic var name, like "matherModules".

File: src/uiveri5.js [hunk 3]
```diff
@@ -217,6 +217,13 @@ function run(config) {
...
+        if(apiTestingMatchers) {
```
Comment (maximnaidenov) [inline src/uiveri5.js]: let's have an empty matchers array in profile.conf.js, just like plugins[]

File: conf/api.profile.conf.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+var profile = require('./profile.conf').config;
+var integrationProfile = require('./integration.profile.conf').config;
+
+integrationProfile.matchers = [
```
Comment (maximnaidenov) [inline conf/api.profile.conf.js]: you can also add those to 'profile' and so avoid the import of integration.profile.conf

File: docs/usage/api.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    .catch(response => {
+        expect(response.status).toBe(401);
+        expect(response.body).toMatch(/access denied/);
+    });
+```
+
+## Save result from call and use it in another call
+```javascript
+let contacts;
+request.get('http://myapi.dev.hana.ondemand.com/contacts/1')
```
Comment (maximnaidenov) [inline docs/usage/api.md]: .do()

File: docs/usage/api.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    .then((res) => {
+        contacts = res.body;
+    });
+
+// the arrow function is necesary to postpone the URL building till the actual execution time
+request.delete('http://myapi.dev.hana.ondemand.com/contacts/1') => `/contacts/{contacts[0].id}`);
```
Comment (maximnaidenov) [inline docs/usage/api.md]: .do()

File: e2e/Runner.js [hunk 7]
```diff
@@ -18,10 +18,16 @@ module.exports = class Runner {
         });
     }
 
-    static startApp(root) {
+    static startApp(root, isMockApp) {
```
Comment (maximnaidenov) [inline e2e/Runner.js]: you can avoid this parameter by inspective the type of the first parameter, _.isString()

File: e2e/scenario/fixture/api.conf.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+exports.config = {
+  profile: 'api',
+  baseUrl: null,
+  matchers: [
```
Comment (maximnaidenov) [inline e2e/scenario/fixture/api.conf.js]: no need to add matchers here, they are already in the profile

**Pull request merged**